File: leetcode/1. Two Sum.ts
//https://leetcode.com/problems/two-sum/description/
// function twoSum(nums: number[], target: number): number[] {
//     const res: number[] = []

//     for(let i = 0; i < nums.length; i++) {
//         for(let j = i+1; j < nums.length; j++) {
//             if(nums[i] + nums[j] === target) res.push(i, j)
//         }
//     }

//     return res
// };
// 

function twoSum(nums: number[], target: number): number[] {
    const map = new Map()
    
    for(let i = 0; i < nums.length; i++) {
        const diff = target - nums[i]
        if(map.has(diff)) {
            return [i, map.get(diff)]
        } else {
            map.set(nums[i], i)
        }
    }
    
    return []
};

File: leetcode/1. Two Sum.go
func twoSum(nums []int, target int) []int {
	m := make(map[int]int)

	for i, v := range nums {

		if pos, ok := m[target-v]; ok {
			return []int{pos, i}
		}
		m[v] = i
	}

	return []int{}
}

func twoSum(nums []int, target int) []int {
	var slice []int

	for i := 0; i < len(nums); i++ {
		for j := i + 1; j < len(nums); j++ {
			if target-nums[i]-nums[j] == 0 {
				slice = append(slice, i, j)
			}
		}
	}

	return slice
}
